frontend/src/main/components/RecommendationRequest/RecommendationRequestTable.js

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import OurTable, { ButtonColumn } from "main/components/OurTable";
+
+import { useBackendMutation } from "main/utils/useBackend";
+import {
+  cellToAxiosParamsDelete,
+  onDeleteSuccess,
+} from "main/utils/RecommendationRequestUtils";
+import { useNavigate } from "react-router-dom";
+import { hasRole } from "main/utils/currentUser";
+
+export default function RecommendationRequestTable({ requests, currentUser }) {
+  const navigate = useNavigate();
+
+  const editCallback = (cell) => {
+    navigate(`/requests/edit/${cell.row.values.id}`);
+  };
+
+  // Stryker disable all : hard to test for query caching
+
+  const deleteMutation = useBackendMutation(
+    cellToAxiosParamsDelete,
+    { onSuccess: onDeleteSuccess },
+    ["/api/recommendationrequest/all"],
+  );
+  // Stryker restore all
+
+  // Stryker disable next-line all : TODO try to make a good test for this
+  const deleteCallback = async (cell) => {
+    deleteMutation.mutate(cell);
+  };
+
+  const columns = [
+    {
+      Header: "id",
+      accessor: "id", // accessor is the "key" in the data
+    },
+    {
+      Header: "Professor Name",
+      accessor: "professorName",
+    },
+    {
+      Header: "Professor Email",
+      accessor: "professorEmail",
+    },
+    {
+      Header: "Requester Name",
+      accessor: "requesterName",
+    },
+    {
+      Header: "Recommendation Type",
+      accessor: "recommendationType",
+    },
+    {
+      Header: "Details",
+      accessor: "details",
+    },
+    {
+      Header: "Status",
+      accessor: "status",
+    },
+    {
+      Header: "Submission Date",
+      accessor: "submissionDate",
+    },
+    {
+      Header: "Completion Date",
+      accessor: "completionDate",
+    },
+  ];
+
+  if (hasRole(currentUser, "ROLE_ADMIN")) {
+    columns.push(
+      ButtonColumn(
+        "Edit",
+        "primary",
+        editCallback,
+        "RecommendationRequestTable",
+      ),
+    );
+    columns.push(
```

Comment:
```suggestion
    }
    if (hasRole(currentUser, "ROLE_USER") ||  hasRole(currentUser, "ROLE_ADMIN"){
    columns.push(
```

Reply:
I updated the permissions for the users and admins for the table and added adequate testing for the new additions to table. In Storybook, the users and admin table shows all three requests from the fixtures. Once the CRUD operation are in place, the data that the admin and user can access will be limited by the GET operation (meaning that the table component be checking which data to return in the table). Is this correct? 

Also as another note, I made an update to the fixtures file that was already merged into main (so that it could the professor's email and name as well as the user's name and email). I updated the fixtures files according to the User entity file. It seems to work fine on storybook (I added some pictures to this comment), was this correct way to approach it? 
user:
![Screenshot 2024-11-24 at 5 45 25 PM](https://github.com/user-attachments/assets/12f44ac0-77a0-474a-aae9-99e8be441a35)
admin: 
![Screenshot 2024-11-24 at 5 45 34 PM](https://github.com/user-attachments/assets/4d7f1306-e6a2-4b0c-a6bf-467f4d5eb186)

Reply:
Yes, I think so!